native-media/app/src/main/java/com/example/nativemedia/NativeMedia.java

```diff
@@ -67,7 +67,8 @@ public void onCreate(Bundle icicle) {
         super.onCreate(icicle);
         setContentView(R.layout.main);
 
-        assetMgr = getResources().getAssets();
+        // get application's assetManager to avoid being garbage-collected.
+        assetMgr = getAssets();
```

Comment:
getAssets() is internally just getResources().getAssets().

This should be getApplication().getAssets().